### PR TITLE
Simplify helpers for containers

### LINKED ISSUE #5404: Code fragment does not build with GCC -O2
```cpp
dev::bytes test(dev::h160 const& addr)
{
    return dev::bytes{0xff} + addr.asBytes();
}
```

```
In file included from /usr/include/string.h:494:0,
                 from /usr/include/c++/7/cstring:42,
                 from /home/chfast/.hunter/_Base/c4cfcc0/af6c4e5/ef3c3eb/Install/include/boost/functional/hash/detail/hash_float.hpp:21,
                 from /home/chfast/.hunter/_Base/c4cfcc0/af6c4e5/ef3c3eb/Install/include/boost/functional/hash/hash.hpp:21,
                 from /home/chfast/.hunter/_Base/c4cfcc0/af6c4e5/ef3c3eb/Install/include/boost/functional/hash.hpp:6,
                 from /home/chfast/Projects/ethereum/aleth/libdevcore/FixedHash.h:30,
                 from /home/chfast/Projects/ethereum/aleth/test/unittests/libdevcore/FixedHash.cpp:18:
In function ‘void* memcpy(void*, const void*, size_t)’,
    inlined from ‘std::vector<_RealType>& dev::operator+=(std::vector<typename std::enable_if<std::is_pod<_T>::value, T>::type>&, const std::vector<_RealType>&) [with T = unsigned char]’ at /home/chfast/Projects/ethereum/aleth/libdevcore/CommonData.h:248:8,
    inlined from ‘std::vector<_RealType> dev::operator+(const std::vector<_RealType>&, const std::vector<_RealType>&) [with T = unsigned char]’ at /home/chfast/Projects/ethereum/aleth/libdevcore/CommonData.h:310:13,
    inlined from ‘dev::bytes test(const h160&)’ at /home/chfast/Projects/ethereum/aleth/test/unittests/libdevcore/FixedHash.cpp:136:44:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:34:71: error: ‘void* __builtin_memcpy(void*, const void*, long unsigned int)’: specified size 18446744073709551615 exceeds maximum object size 9223372036854775807 [-Werror=stringop-overflow=]
   return __builtin___memcpy_chk (__dest, __src, __len, __bos0 (__dest));
                                                                       ^
```

### PR DESCRIPTION
**Pull request by chfast**
Fixes #5404

### CODE REVIEW

File: libdevcore/CommonData.h [hunk 1]
```diff
@@ -302,17 +281,6 @@ template <class T, class U> std::vector<T> operator+(std::vector<T> _a, U const&
...
-template <class T>
-inline std::vector<T> operator+(std::vector<T> const& _a, std::vector<T> const& _b)
-{
-	std::vector<T> ret(_a);
-	return ret += _b;
-}
-
-
-/// Make normal string from fixed-length string.
-std::string toString(string32 const& _s);
```
Comment (gumb0) [inline libdevcore/CommonData.h]: This `string32` from `libdevcore/Common.h` looks to be not used anywhere

Comment (chfast) [reply libdevcore/CommonData.h]: That's the reason for removal

Comment (gumb0) [reply libdevcore/CommonData.h]: But you didn't remove `string32` itself?

Comment (chfast) [reply libdevcore/CommonData.h]: Mistake

**Pull request merged**